develop/components/common/SideBar.tsx

```diff
@@ -1,43 +1,78 @@
+        }
+    }
 
     return (
         <div className={"side-bar" + (props.toggle ? " open" : "")}>
             <div className="category">
                 <div className="title">How it works?</div>
                 <div className="nav-holder">
-                    <Link to="/about" className={(page === "/" || page === "/about") ? "active" : ""}><span>About</span></Link>
+                    <a href="#/about" onClick={navigate} className={(page === "/" || page === "/about") ? "active" : ""}><span>About</span></a>
                     <a href="https://github.com/sebgroup/react-components/issues" target="_blank"><span>Issues</span></a>
                     <a href="https://github.com/sebgroup/react-components/releases" target="_blank"><span>Release notes</span></a>
                 </div>
             </div>
             <div className="category">
                 <div className="title">Components</div>
                 <div className="nav-holder">
-                    <div className="sub-title">Form</div>
-                    {forms && forms.map((item, index) =>
-                        <Link key={index} to={item.path} className={page === item.path ? "active" : ""}><span>{item.name}</span></Link>
+                    {forms && <div className="sub-title">Form</div>}
+                    {forms && forms.map((item: SideBarItem, index: number) =>
+                        <a
+                            key={index}
+                            href={"#" + item.path}
+                            onClick={navigate}
+                            className={page === item.path ? "active" : null}
+                        >
+                            <span>{item.name}</span>
+                        </a>
                     )}
-                    <div className="sub-title">UI</div>
-                    {uis && uis.map((item, index) =>
-                        <Link key={index} to={item.path} className={page === item.path ? "active" : ""}><span>{item.name}</span></Link>
+                    {uis && <div className="sub-title">UI</div>}
+                    {uis && uis.map((item: SideBarItem, index: number) =>
+                        <a
+                            key={index}
+                            href={"#" + item.path}
+                            onClick={navigate}
+                            className={page === item.path ? "active" : null}
+                        >
+                            <span>{item.name}</span>
+                        </a>
                     )}
-                    <div className="sub-title">Other</div>
-                    {others && others.map((item, index) =>
-                        <Link key={index} to={item.path} className={page === item.path ? "active" : ""}><span>{item.name}</span></Link>
+                    {others && <div className="sub-title">Other</div>}
+                    {others && others.map((item: SideBarItem, index: number) =>
+                        <a
```

Comment:
I recommend using react-router [Link](https://knowbody.github.io/react-router-docs/api/Link.html)  than a anchor. The anchor here often times triggers refresh, reload or remounting when in the same path. Although React router Link translate to a anchor, they handle this issue for you behind the scene.

Reply:
I could, but when I was refactoring I noticed that this component doesn't cater for the case where you are trying to navigate to same URL you are in which throws a warning in the console. So when I added an `onClick` method and prevent the default to run the check, I saw no need to have it.